R/utils.R

```diff
@@ -189,25 +189,46 @@ check_filter_datanames <- function(filters, datanames) {
 #' @param modules (`teal_modules`) object
 #' @param filters (`teal_slices`) object
 #' @param filtered_data_singleton A result of `teal_data_to_filtered_data` applied to `data`.
+#' @param progress (`Progress`) object, optional; will be filled
 #' @return Returns list of same shape as `modules`, containing `FilteredData` at every leaf.
 #' If module specific, each leaf contains different instance, otherwise every leaf contains `filtered_data_singleton`.
 #' @keywords internal
-modules_datasets <- function(data, modules, filters, filtered_data_singleton = teal_data_to_filtered_data(data)) {
+modules_datasets <- function(data,
+                             modules,
+                             filters,
+                             filtered_data_singleton = teal_data_to_filtered_data(data),
+                             progress) {
   checkmate::assert_class(data, "teal_data")
   checkmate::assert_multi_class(modules, c("teal_modules", "teal_module"))
   checkmate::assert_class(filters, "modules_teal_slices")
   checkmate::assert_r6(filtered_data_singleton, "FilteredData")
+  if (!missing(progress)) checkmate::assert_r6(progress, "Progress")
 
   if (!isTRUE(attr(filters, "module_specific"))) {
+    if (!missing(progress)) {
```

Comment:
why and when `progress` can be missing, if it's created in `[module_teal.R](https://github.com/insightsengineering/teal/pull/1236/files#diff-7b625abea8aaccebc90e0fddf962570dbeb41b6b470c7fed09892c6c88094b17)`?

Reply:
Just in case the argument isn't provided but I guess if it's an internal function that will only ever be called here, it can be assumed that the argument is always there. I will remove the condition.

Reply:
It's internal - let's remove, so there is less clutter in the code.

Reply:
done

Reply:
Ah, I remember now. If the argument is mandatory, unit tests fail. Would you like me to include a `Progress` object in the tests?

Reply:
And here is why I didn't:
`Error in `initialize(...)`: Can only use Progress$new() inside a Shiny app`

Reply:
coolio